storage/innobase/xtrabackup/src/utils.cc

```diff
@@ -100,5 +107,39 @@ unsigned long get_version_number(std::string version_str) {
   return major * 10000 + minor * 100 + version;
 }
 
+#ifdef __APPLE__
+unsigned long free_memory() {
+  int64_t unused_mem;
+  vm_size_t page_size;
+  mach_msg_type_number_t count;
+  vm_statistics_data_t vm_stats;
+  mach_port_t host = mach_host_self();
+  count = sizeof(vm_stats) / sizeof(natural_t);
+  if (KERN_SUCCESS == host_page_size(host, &page_size) &&
+      KERN_SUCCESS ==
+          host_statistics(host, HOST_VM_INFO, (host_info_t)&vm_stats, &count)) {
+    unused_mem = (int64_t)vm_stats.free_count * (int64_t)page_size;
+    return (unsigned long)unused_mem;
+  }
+  return 0;
+}
+
+unsigned long total_memory() {
+  unsigned long total_mem = sysconf(_SC_PHYS_PAGES) * sysconf(_SC_PAGESIZE);
+  return total_mem;
+}
+
+#else
+unsigned long free_memory() {
+  meminfo();
+  return kb_main_available * 1024;
+}
+
+unsigned long total_memory() {
+  meminfo();
+  return kb_main_total * 1024;
+}
+#endif
+
```

Comment:
better to prefix host_ or system_ to avoid confusion with buffer pool total free avail memory?